### ISSUE TITLE
Issue running on Apple Silicon ARM architecture

### ISSUE THREAD

**Issue by fbordwell** [vyrjana/DearEIS#4]
**Desktop (please complete the following information):**
- _DearEIS_ version [e.g. 0.1.0]: 3.1.3
- Python version [e.g. 3.9]: 3.10
- OS: [e.g. Linux, Windows, macOS]: macOS

**Describe the bug**
The program is unable to be ran when installing through `pip` as the `cvxopt` package does not run on Apple Silicon hardware. This is documented in [issue 195](https://github.com/cvxopt/cvxopt/issues/195) of `cvxopt`. A [fork](https://github.com/sanurielf/kvxopt) claims to fix this issue.

**Screenshots/videos/tracebacks**
```
~/Programming/thirdparty/DearEIS $ deareis
Traceback (most recent call last):
  File "/opt/homebrew/bin/deareis", line 5, in <module>
    from deareis.program import main
  File "/opt/homebrew/lib/python3.10/site-packages/deareis/__init__.py", line 24, in <module>
    from deareis.data import Project
  File "/opt/homebrew/lib/python3.10/site-packages/deareis/data/__init__.py", line 20, in <module>
    from .data_sets import DataSet
  File "/opt/homebrew/lib/python3.10/site-packages/deareis/data/data_sets.py", line 25, in <module>
    import pyimpspec
  File "/opt/homebrew/lib/python3.10/site-packages/pyimpspec/__init__.py", line 48, in <module>
    from pyimpspec.analysis import (
  File "/opt/homebrew/lib/python3.10/site-packages/pyimpspec/analysis/__init__.py", line 31, in <module>
    from .drt import (
  File "/opt/homebrew/lib/python3.10/site-packages/pyimpspec/analysis/drt/__init__.py", line 27, in <module>
    from .tr_rbf import _calculate_drt_tr_rbf
  File "/opt/homebrew/lib/python3.10/site-packages/pyimpspec/analysis/drt/tr_rbf.py", line 82, in <module>
    from cvxopt import (
  File "/opt/homebrew/lib/python3.10/site-packages/cvxopt/__init__.py", line 50, in <module>
    import cvxopt.base
ImportError: dlopen(/opt/homebrew/lib/python3.10/site-packages/cvxopt/base.cpython-310-darwin.so, 0x0002): tried: '/opt/homebrew/lib/python3.10/site-packages/cvxopt/base.cpython-310-darwin.so' (mach-o file, but is an incompatible architecture (have (x86_64), need (arm64e)))
```

**Comment by vyrjana** [comment]
Have you checked if `kvxopt` works on your system (i.e., you can install it and at least import it in the Python interpreter)? As far as I can tell GitHub Actions still only provides x86-64 systems for use with macOS runners so I cannot readily test on Apple Silicon hardware.

I'm a bit wary about completely switching over to a fork since `cvxopt` has not been abandoned. However, I could add support for `kvxopt` as an optional dependency and refactor the code so that it tries to import `kvxopt` if `cvxopt` cannot be imported. How does that sound?

**Comment by fbordwell** [comment]
Thanks for the quick response. `kvxopt` works for importing at least. I look forward to trying the program when the changes are implemented.

```
~/Programming/thirdparty/DearEIS $ poetry run python -c "import cvxopt"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/fischer/Library/Caches/pypoetry/virtualenvs/deareis-k_AUrKUV-py3.10/lib/python3.10/site-packages/cvxopt/__init__.py", line 50, in <module>
    import cvxopt.base
ImportError: dlopen(/Users/fischer/Library/Caches/pypoetry/virtualenvs/deareis-k_AUrKUV-py3.10/lib/python3.10/site-packages/cvxopt/base.cpython-310-darwin.so, 0x0002): tried: '/Users/fischer/Library/Caches/pypoetry/virtualenvs/deareis-k_AUrKUV-py3.10/lib/python3.10/site-packages/cvxopt/base.cpython-310-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/fischer/Library/Caches/pypoetry/virtualenvs/deareis-k_AUrKUV-py3.10/lib/python3.10/site-packages/cvxopt/base.cpython-310-darwin.so' (no such file), '/Users/fischer/Library/Caches/pypoetry/virtualenvs/deareis-k_AUrKUV-py3.10/lib/python3.10/site-packages/cvxopt/base.cpython-310-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))
~/Programming/thirdparty/DearEIS $ poetry run python -c "import kvxopt"
~/Programming/thirdparty/DearEIS $ 
```

**Comment by vyrjana** [comment]
I just published a smaller update (version 3.1.3) for `pyimpspec` that should fix the issue. So grab the latest version of `pyimpspec` and you should be good to go. Let me know if you run into any issues.

**Comment by fbordwell** [comment]
That fixes the problem in my limited testing. Thanks!
